Makefile

```diff
@@ -19,19 +19,33 @@ sync_artifacts:
 	aws s3 sync models/ s3://$(PRIVATE_PROJECT_BUCKET)/models/
 	aws s3 sync s3://$(PRIVATE_PROJECT_BUCKET)/models/ models/
 
-.PHONY:virtualenv
-virtualenv:
+$(VIRTUALENV)/.installed:
 	@if [ -d $(VIRTUALENV) ]; then rm -rf $(VIRTUALENV); fi
 	@mkdir -p $(VIRTUALENV)
 	virtualenv --python python3 $(VIRTUALENV)
 	$(VIRTUALENV)/bin/pip install -r requirements.txt
 	$(VIRTUALENV)/bin/pip install -r requirements_test.txt
 	$(VIRTUALENV)/bin/pip install -e .
+	touch $@
+
+.PHONY: virtualenv
+virtualenv: $(VIRTUALENV)/.installed
+
+.PHONY: virtualenv_scispacy
+virtualenv_scispacy: virtualenv
+	$(VIRTUALENV)/bin/pip install -r requirements_scispacy.txt
+	rm $(VIRTUALENV)/.installed
+	echo "⚠️  Virtualenv is broken and needs to be reinstalled"
 
 .PHONY: test
-test:
+test: virtualenv
 	$(VIRTUALENV)/bin/python -m spacy download en_trf_bertbaseuncased_lg
-	$(VIRTUALENV)/bin/pytest --disable-warnings -v --cov=grants_tagger
+	$(VIRTUALENV)/bin/pytest --disable-warnings -v --cov=grants_tagger -m "not scispacy"
+
+.PHONY: test_scispacy
```

Comment:
FYI when I did `make test_scispacy` I got 
```
ERROR: Command errored out with exit status 128: git clone -q git://github.com/wellcometrust/WellcomeML.git /Users/gallaghe/Code/grants_tagger/venv/src/wellcomeml Check the logs for full command output.
```
Could you just have `wellcomeml[deep-learning]` in the requirements now or did you need that specific wellcomeml version?

When I did replace this with `wellcomeml[deep-learning]` then `make test_scispacy` passed

Reply:
This is a good point, the version is pinned to have reproducible results. But in order to run this code you need to unpin it indeed. I think I should have just mentioned it in the description. It is intentionally as is.

Reply:
@lizgzil are you sure you runned `make test_scispacy`? do you mind running again as it passes locally and in the tests.

Reply:
it looks like that error was actually when I ran `make test` 🤦‍♀️  , `make test_scispacy` did pass (so will approve), but I got the error/warning:
```
ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.

wellcomeml 2020.9.0 requires spacy==2.2.1, but you'll have spacy 2.3.2 which is incompatible.
spacy-transformers 0.5.1 requires spacy<2.3.0,>=2.2.1, but you'll have spacy 2.3.2 which is incompatible.
```
so maybe you'll need to do that change soon

Reply:
in order to run make test you need to recreate the environment as the scispacy installation breaks it. Check https://github.com/wellcometrust/grants_tagger/blob/1e8ff1e04722f6c431368127fca2fd321bcb9afc/Makefile#L38